src/components/SignInForm.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { FieldError, useForm, Controller } from 'react-hook-form';
+
+import { Input } from './Input';
+
+export interface ISignInData {
+  email: string;
+  password: string;
+}
+
+export interface IErrorData {
+  email?: FieldError | undefined;
+  password?: FieldError | undefined;
+}
+
+export const SignInForm: React.FC = () => {
+  const {
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm<ISignInData>({ defaultValues: { password: '' } });
+  const handleRegistration = (data: ISignInData) => console.log('data', data);
+  const handleError = (data: IErrorData) => {
+    console.log(data);
+  };
+
+  const registerOptions = {
+    name: { required: 'Name is required' },
+    email: { required: 'Email is required' },
+    password: {
+      required: 'Password is required',
+      minLength: {
+        value: 8,
+        message: 'Password must have at least 8 characters',
+      },
+    },
+  };
+
+  return (
+    <form
+      className="flex flex-col justify-between items-center"
+      onSubmit={handleSubmit(handleRegistration)}
+    >
+      <div className="flex flex-col justify-center items-center">
+        <Controller
+          name="email"
+          control={control}
+          rules={{ required: true }}
+          render={({ field }) => (
+            <Input
+              label={'email'}
+              errors={errors.email ? registerOptions.email.required : null}
+              {...field}
+            />
+          )}
+        />
+      </div>
+      <div className="flex flex-col justify-center items-center">
+        <Controller
```

Comment:
It's great that you tried different API's of react-hook-form. But most of the time people use {...register} with such a simple input component. We will try Controller with DatePicker and something more difficult in the future. Notice how we can write a validation using react-hook-form, in the example below
```
       <Input
          {...register('password', {
            required: 'Password is required',
            minLength: { value: 8, message: 'Password must have at least 8 characters' },
          })}
          label="Password"
          errors={errors?.password?.message ?? null}
        />
```